### PR TITLE
Imports changed datasets from live where change <24 hours ago.

### LINKED ISSUE #236: Daily import
We should build a task that can be run every X hours that will import any legacy datasets that have changed in the last X hours.  We should refuse to import if it has been changed on publish as well.

### PR DESCRIPTION
**Pull request by rossjones**
Adds a rake task to import datasets which have been modified in the
last 24 hours (by asking live DGU for the metadata).

Fixes #236

### CODE REVIEW

File: lib/util/metadata_tools.rb [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    d = Dataset.find_or_create_by(uuid: obj["id"])
+    d.name = obj["name"]
+    d.title = obj["title"]
+    d.summary = generate_summary(obj["notes"])
+    d.description = obj["notes"]
+    d.organisation_id = orgs_cache[obj["owner_org"]]
+    d.licence = obj["license_id"]
+    d.licence = 'no-licence' if d.licence == ""
+    d.licence_other = ""
+    #d.frequency = convert_frequency(obj["update_frequency"])
```
Comment (hannako) [inline lib/util/metadata_tools.rb]: is this mean to be commented out, and is setting them all to `never` right?

File: lib/util/metadata_tools.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  def add_dataset_metadata(obj, orgs_cache)
+    d = Dataset.find_or_create_by(uuid: obj["id"])
+    d.name = obj["name"]
+    d.title = obj["title"]
+    d.summary = generate_summary(obj["notes"])
+    d.description = obj["notes"]
+    d.organisation_id = orgs_cache[obj["owner_org"]]
+    d.licence = obj["license_id"]
+    d.licence = 'no-licence' if d.licence == ""
+    d.licence_other = ""
```
Comment (govuklaurence) [inline lib/util/metadata_tools.rb]: ^ agreed

Comment (rossjones) [inline lib/util/metadata_tools.rb]: There's a diff between no-licence and other :) We don't know what it is if it isn't specified so can't set to other.  This does need further work though for other licences we do know about.

File: lib/util/metadata_tools.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      inspire.save!()
+    end
+
+    # Iterate over the resources list and add a new datafile for each
+    # item.
+    obj["resources"].each do |resource|
+      add_resource(resource, d)
+    end
+
+    d.published = true
```
Comment (govuklaurence) [inline lib/util/metadata_tools.rb]: i think it's assuming that everything in the import IS published, which may be true - @rossjones ?

File: lib/util/metadata_tools.rb (hunk 1 above)
Comment (rossjones) [inline lib/util/metadata_tools.rb]: Yes, it's not right, we need to do more in another PR.

File: lib/util/metadata_tools.rb [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    inspire.spatial_data_service_type = get_extra(extras, 'spatial-data-service-type')
+    inspire.spatial_reference_system = get_extra(extras, 'spatial-reference-system')
+    inspire
+  end
+
+  # Generates a summary from the provided notes as in
+  # the legacy metadata we don't have a summary field.
+  def generate_summary(notes)
+    return notes if notes && notes != ""
+    "No description provided"
```
Comment (rossjones) [inline lib/util/metadata_tools.rb]: It's handled by the view, but we could also do it here.

File: lib/util/metadata_tools.rb [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  end
+
+  # Returns a value if the dataset is harvested, i.e. it has
+  # a harvest object id.
+  def harvested?(obj)
+    get_extra(obj["extras"], "harvest_object_id") != ""
+  end
+
+  # Determine whether datafile is documentation or not
+  def get_doc_type(fmt)
```
Comment (rossjones) [inline lib/util/metadata_tools.rb]: Yeah, will fix in the next cleanup PR.
